nixos/modules/services/misc/home-assistant.nix

```diff
@@ -38,6 +38,11 @@ let
     then (cfg.package.override { inherit extraComponents; })
     else cfg.package;
 
+  defaultConfig = {
+    homeassistant.time_zone = config.time.timeZone;
+    http.server_port = (toString cfg.port);
```

Comment:
Note that this will enable the http component for everybody with `config != null`. Not so big an issue as just about everyone enables it anyway. But we might want to add an option for not setting the default config.

Reply:
I've added ```applyDefaultConfiguration```.